backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Context.kt

```diff
@@ -112,20 +113,31 @@ internal class SpecialDeclarationsFactory(val context: Context) {
         assert(overriddenFunction.needBridge) {
             "Function ${irFunction.descriptor} is not needed in a bridge to call overridden function ${overriddenFunction.overriddenFunction.descriptor}"
         }
-        val bridgeDirections = overriddenFunction.bridgeDirections
-        return bridgesDescriptors.getOrPut(irFunction to bridgeDirections) {
-            createBridge(irFunction, bridgeDirections)
-        }
+        val key = BridgeKey(irFunction, overriddenFunction.bridgeDirections)
+        return bridges.getOrPut(key) { createBridge(key) }
     }
 
-    private fun createBridge(function: IrSimpleFunction,
-                             bridgeDirections: BridgeDirections) = WrappedSimpleFunctionDescriptor().let { descriptor ->
+    private fun createBridge(key: BridgeKey) = WrappedSimpleFunctionDescriptor().let { descriptor ->
+        val (function, bridgeDirections) = key
         val startOffset = function.startOffset
         val endOffset = function.endOffset
-        val returnType = when (bridgeDirections.array[0]) {
+        val returnType = when (val returnBridgeDirection = bridgeDirections.array[0]) {
```

Comment:
How is this essentially different from `function.returnType`? Could you provide and example?

Reply:
Here `function` is not the function being overridden, but the target to call. So exactly for targets returning `Nothing` or `Nothing?` return type is different.

Reply:
Ah, I see. Then:
How is this essentially different from `overriddenFunction.returnType`? Could you provide an example?
Do I understand correctly that `overriddenFunction` is not currently available here, and this is the only blocker?

Reply:
Yes, I think, `overriddenFunction.returnType` will do, I guess, it is not available here for the same reason - I wanted a bridge to be built only by its target and directions.

Reply:
Of course.
But it might be possible to include the type (or information to construct it) to `BridgeDirection`.
This might involve using result of https://github.com/JetBrains/kotlin-native/blob/7a24e9ea0ecb7a8f758ae13fa8f3186700c7bfe9/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Boxing.kt#L27 for example.